packages/antify-admin/src/server/api/roles/[roleId].delete.ts

```diff
@@ -16,9 +17,28 @@ export default defineEventHandler(async (event) => {
     throw new HttpForbiddenError();
   }
 
+  const access = await prisma.userTenantAccess.findMany({
+    select: {
+      tenantId: true,
+      userId: true,
+      roleId: true,
+    },
+    where: {
+      roleId: event.context.params.roleId,
+    },
+  });
+
+  if (access && access.length > 0) {
```

Comment:
This query and condition is used on multiple places. Please outsorce into a service.